src/commands/init.ts

```diff
@@ -1,57 +1,117 @@
+    }
+
     try {
-      const kubernetesVersion = await promptForKubernetesVersion();
+      let kubernetesVersion = schemaVersion;
+      let frameworks = framework ? [framework] : [];
 
-      if (!isDefined(kubernetesVersion)) {
-        return;
-      }
+      if (interactive) {
+        kubernetesVersion = await promptForKubernetesVersion();
 
-      const frameworks = await promptForFrameworks();
+        if (!isDefined(kubernetesVersion)) {
+          return;
+        }
 
-      if (!isDefined(frameworks)) {
-        return;
-      }
+        frameworks = await promptForFrameworks();
+
+        if (!isDefined(frameworks)) {
+          return;
+        }
 
-      const configs: Config[] = await Promise.all(frameworks.map(async (framework: Framework) => {
-        return getFrameworkConfig(framework);
-      }));
+        if (hasConfig) {
+          const canOverwrite = await promptForOverwrite();
 
-      const fullConfig: Config = {
-        plugins: configs.reduce<PluginMap>((acc, config) => {
-          return {
-            ...acc,
-            ...config.plugins
+          if (!canOverwrite) {
+            return;
           }
-        }, {}),
-        rules: mergeRules(configs.map(config => config.rules).filter(isDefined)),
-        settings: {
+        }
+      }
+
+      const config: Config = {};
+      if (frameworks.length > 0) {
+        const configs: Config[] = (await Promise.all(frameworks.map(async (framework: Framework) => {
+          return getFrameworkConfig(framework);
+        }))).filter(isDefined);
+
+        const fullConfig: Config = {
+          plugins: configs.reduce<PluginMap>((acc, config) => {
+            return {
+              ...acc,
+              ...config.plugins
+            }
+          }, {}),
+          rules: mergeRules(configs.map(config => config.rules).filter(isDefined)),
+        };
+
+        config.plugins = fullConfig.plugins;
+        config.rules = fullConfig.rules;
+      } else {
+        const defaultConfigJson = defaultConfig as Config;
+        config.plugins = defaultConfigJson.plugins;
+        config.rules = defaultConfigJson.rules;
+        config.settings = defaultConfigJson.settings;
+      }
+
+      if (isDefined(kubernetesVersion)) {
+        config.settings = {
           "kubernetes-schema": {
-            schemaVersion: kubernetesVersion,
+            schemaVersion: `v${kubernetesVersion}`.replace(/v+/g, 'v'),
           },
-        },
-      };
+        };
+      }
 
       const configYaml = new Document();
-      (configYaml.contents as any) = fullConfig;
-      await writeFile(resolve('monokle.validation.yaml'), configYaml.toString())
-
-      print(success(resolve('monokle.validation.yaml')));
+      (configYaml.contents as any) = config;
+      await writeFile(configPath, configYaml.toString())
 
+      print(success(configPath));
     } catch (err: any) {
       print(error(err.message));
```

Comment:
as above..

Reply:
I'll change this to throw here since it relates to automatic scenario too 👍